pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,120 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_terminators():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig = Figure()
+    fig.basemap(region="d", projection="W0/15c", frame="a")
+    fig.solar(
+        terminator="d",
+        pen="1p,blue",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="a",
+        pen="1p,red",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="c",
+        pen="1p,green",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        terminator="n",
+        pen="1p,yellow",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare(filename="test_solar_set_terminator_datetime.png")
+@pytest.mark.parametrize(
+    "terminator_datetime",
+    [
+        "1990-02-17 04:25:00",
```

Comment:
```suggestion
        "1990-02-17T04:25:00",
```
@seisman, you can just put a 'T' here to avoid the space in the filename.

Reply:
I know, but these two are different. `1990-02-17T04:25:00` is directly supported by GMT, but `1990-02-17 04:25:00` must be converted to a datatime object by pandas before passing to GMT.

Reply:
My reading of the code implementation is that the argument into `terminator_datetime` always goes through `pd.to_datetime` though? So you want to test a string that is not supported by GMT but supported by Pandas?

https://github.com/GenericMappingTools/pygmt/blob/b8edbdc9a799de85032b88308d6195bb3ee6b592/pygmt/src/solar.py#L88-L95

Reply:
Interesting, the test still fails on Windows because the filename (I think with colons ':') doesn't work even without spaces https://github.com/GenericMappingTools/pygmt/runs/2189106570?check_suite_focus=true#step:11:488:

```python-traceback
___________ test_solar_set_terminator_datetime[1990-02-17T04:25:00] ___________

self = WindowsPath('D:/a/pygmt/pygmt/tmp-test-dir-with-unique-name/results/pygmt.tests.test_solar.test_solar_set_terminator_datetime_1990-02-17T04:25:00')
mode = 511, parents = True, exist_ok = True

    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
        """
        Create a new directory at this given path.
        """
        if self._closed:
            self._raise_closed()
        try:
>           self._accessor.mkdir(self, mode)
E           OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'D:\\a\\pygmt\\pygmt\\tmp-test-dir-with-unique-name\\results\\pygmt.tests.test_solar.test_solar_set_terminator_datetime_1990-02-17T04:25:00'
```

Reply:
Should be resolved with https://github.com/GenericMappingTools/pygmt/pull/804#discussion_r600968043.